### ISSUE TITLE
[7.5.0-RC5] MetadataStore and GigaChannelManager shutdown sequence error

### ISSUE THREAD

**Issue by devos50** [Tribler/tribler#5297]
Found this in the log when shutting down Tribler. Even though it did not crash Tribler, it might be the case that the channel processing procedure is not shut down correctly (since the metadata store is already set to `None` while the channel is still processing).

```
[PID:4931] 2020-04-22 10:46:54,838 - ERROR <gigachannel_manager:266> GigaChannelManager._process_download(): Error when processing channel dir download: 'NoneType' object has no attribute '_db'
```

**Comment by ichorid** [comment]
Indeed, this points to a bigger problem of shutdown sequence being a mess between GigaChannelManager and MetadataStore. To solve it properly, the channel-processing and thread-handling functionality should be arranged between them differently. I'm not sure this is worth it at RC5 stage.

**Comment by ichorid** [comment]
Most probably solved by #6335 .
